File: 3396-valid-word/3396-valid-word.java
class Solution {
    public boolean isValid(String word) {
        if (word.length()<3) return false;

        for(char c: word.toCharArray()){
            if (!Character.isLetterOrDigit(c)) return false;
        }
        String vowels="aeiouAEIOU";
        boolean hasVowels=false;
        boolean hasConsonant=false;
        for (char c: word.toCharArray()){
            if(Character.isLetter(c)){
                if(vowels.indexOf(c)!=-1){
                    hasVowels=true;
                }else{
                    hasConsonant=true;
                }
            }
            if (hasVowels && hasConsonant) return true;
        }
        return false;
        
    }
}

File: 3396-valid-word/3396-valid-word.py
class Solution:
    def isValid(self, word: str) -> bool:
        if len(word)<3:
            return False
        if not all(c.isalnum() for c in word):
            return False
        vowels=set("aeiouAEIOU")
        has_vowels=any(c in vowels for c in word)
        has_consonant=any(c.isalpha() and c not in vowels  for c in word)
        return has_vowels and has_consonant
        